### ISSUE TITLE
PROF vocabulary terms needed for indicating classes constrained

### ISSUE THREAD

**Issue by nicholascar** [w3c/dx-prof#44]
Currently there is no way, in PROF, to say:

> Class X has constraining elements in Profile Y

How do we say `<Profile_Y> constrains <Class_X>` ?

The motivation for this is that we want to be able to do a reveres look up to know what Profiles (Specifications) are relevant to what classes so that people can know, per class, what Profiles they should consider.

Suggested properties could be:

```turtle
prof:constrains
    a owl:ObjectProperty ;
    rdfs:domain prof:Profile ;
    rdfs:range owl:Class ;  # or perhaps rdf:Resource
    rdfs:comment "A Profile constrains, that is places restrictions on the use of, a Class"@en ;  # ...of, a Resource
    rdfs:label "constrains"@en ;
.

prof:isConstrainedBy
    a owl:ObjectProperty ;
    owl:inverseOf prof:constrains ;
    rdfs:label "is constrained by"@en ;
.
```

**Comment by kcoyle** [comment]
@nicholascar Can you give some examples of classes being constrained? Thanks.

**Comment by nicholascar** [comment]
Certainly: the _Australian Government Ontology Profile_ (AGOP: <https://linked.data.gov.au/def/agop>) requires that ontologies published by the [Australian Government Linked Data Working Group](https://linked.data.gov.au/) have certain metadata present. This is enforced by the _specification_ resource within that profile (<https://linked.data.gov.au/def/agop/spec>) that mandates that instances of `owl:Ontology` have `skos:prefLabel`, `dcterms:created` properties with certain values. The _specification_ 's rules are testable against instance data (RDF files containing ontologies) by a _validation_ resource written in SHACL.

So the constrained class here is `owl:Ontology`: for an instance of such to be valid according to AGOP, it must have such and such properties. Following the logic in this issue's description, we might say:

```turtle
<https://linked.data.gov.au/def/agop> constrains owl:Ontology
```

Similarly, the [VocPub Profile](https://w3id.org/profile/vocpub) constrains `skos:ConceptScheme` but it also `skos:Collection` and `skos:Concept`...

**Comment by leskneebone** [comment]
If it is a specification that constrains a class / resource instance, that is not the Profile but a Profile resource. So yes, it's useful to be able to walk back from a class, property or resource and know where it is constrained by some spec / standard. So does this work?

```turtle
owl:Ontology prof:isConstrainedBy <https://linked.data.gov.au/def/agop/spec> 
```
